mod/state-transition/pkg/core/state_processor_staking.go

```diff
@@ -191,16 +191,38 @@ func (sp *StateProcessor[
 		math.Gwei(sp.cs.MaxEffectiveBalance()),
 	)
 
-	// TODO: This is a bug that lives on bArtio. Delete this eventually.
-	if sp.cs.DepositEth1ChainID() == bArtioChainID {
-		if err := st.AddValidatorBartio(val); err != nil {
+	// BeaconKit enforces a cap on the validator set size. If a deposit is made
+	// that would breach the cap, we mark the validator as immediately
+	// withdrawable, so that it will be evicted in the next block along with
+	// the amount deposited.
+	validators, err := st.GetValidators()
+	if err != nil {
+		return err
+	}
+
+	//#nosec:G701 // can't overflow.
+	if uint32(len(validators)) >= sp.cs.GetValidatorsSetCapSize() {
+		var slot math.Slot
+		slot, err = st.GetSlot()
+		if err != nil {
 			return err
 		}
-	} else if err := st.AddValidator(val); err != nil {
+
+		epoch := sp.cs.SlotToEpoch(slot)
+		val.SetWithdrawableEpoch(epoch)
+	}
+
+	// TODO: This is a bug that lives on bArtio. Delete this eventually.
+	if sp.cs.DepositEth1ChainID() == bArtioChainID {
+		return st.AddValidatorBartio(val)
+	}
```

Comment:
replicating here the fix from https://github.com/berachain/beacon-kit/pull/2111 to avoid double counting Bartio Validators balances (done implicitly in AddValidatorBartio)